go-controller/pkg/cni/cniserver_linux.go

```diff
@@ -15,16 +15,11 @@ import (
 )
 
 // Start the Server's local HTTP server on a root-owned Unix domain socket.
-// requestFunc will be called to handle pod setup/teardown operations on each
+// handlePodRequestFunc will be called to handle pod setup/teardown operations on each
 // request to the Server's HTTP server, and should return the response bytes,
 // or an error when the operation has completed.
-func (s *Server) Start(requestFunc cniRequestFunc) error {
-	if requestFunc == nil {
```

Comment:
looks like this check was here for no reason?

Reply:
when this function is passed to Start, there is a possibility to pass nil.
now it is set on creation in `NewCNIServer`, but can be overriden in 
```
func TestCNIServer(t *testing.T) {
...
        // override request handler
	s.handlePodRequestFunc = serverHandleCNI
}
```